amundsen_application/api/search/v0.py

```diff
@@ -111,114 +147,117 @@ def _create_url_with_field(*, search_term: str, page_index: int) -> str:
     return url
 
 
-@action_logging
-def _search_user(*, search_term: str, page_index: int) -> Dict[str, Any]:
+@search_blueprint.route('/table_qs', methods=['POST'])
+def search_table_query_string() -> Response:
     """
-    call the search service endpoint and return matching results
-    :return: a json output containing search results array as 'results'
-
-    Schema Defined Here:
-    https://github.com/lyft/amundsensearchlibrary/blob/master/search_service/api/user.py
-    TODO: Define an interface for envoy_client
+    TODO (ttannis): Update this docstring after amundsensearch documentation is merged
+    Calls the search service to execute a search. The request data is transformed
+    to the json payload defined [link]
     """
+    request_json = request.get_json()
 
-    def _map_user_result(result: Dict) -> Dict:
-        user_result = dump_user(load_user(result))
-        user_result['type'] = 'user'
-        return user_result
+    search_term = get_query_param(request_json, 'term', '"term" parameter expected in request data')
+    page_index = get_query_param(request_json, 'pageIndex', '"pageIndex" parameter expected in request data')
+    filters = request_json.get('filters', {})
```

Comment:
An earlier version of this allowed users to to hit enter with an empty search term and go to the search page. But I found it created too many edge cases in regards to the application state and url management.

This current version has support that will allow for the same end result in a way that I found easier to manage. Users have a way to go directly to the search page in a way that informs them filter features are available, and when on that page a search will execute with an empty search term if they modify filters but don't enter a term.

Also the empty search page will show some help text, which came from Wiebke's UI review:
![image](https://user-images.githubusercontent.com/1790900/74187743-0e319f00-4c02-11ea-9107-a1dd2622de9d.png)

Reply:
I'm okay with only having the `Advanced Search` here. It might be nice to revisit and tackle those edge cases in a future commit as I think pressing `enter` on an empty search bar would be easier than clicking.

Reply:
I can see that, but I think there's other concepts to consider besides saving clicks:
1. The main purpose of this was to make filters discoverable in the `HomePage`. The breadcrumb achieves that better than users discovering hitting enter will take them to the search page.
2. It's not common behavior for a search bar (see the other popular search UIs we've referenced for evaluating user experience), and I think conceptually it's nice to have components be responsible for only what they need to be responsible to avoid maintaining too much branching logic for various different cases. In this case it would add more branching logic to areas in the redux layer in `submitSearch` and our url updating logic just to achieve a navigation to `/search` -- yet we have a whole component whose only purpose is to execute navigations and so I really couldn't justify complicating our code in that manner.